### PR TITLE
Attempt at fixing the `ConstructASCIIGeometry` helper

### LINKED ISSUE #1145: FairModule::ConstructASCIIGeometry: "warning: parameter 'dataType1' set but not used"
```console
base/sim/FairModule.h:170:44: warning: parameter 'dataType1' set but not used [-Wunused-but-set-parameter]
 void FairModule::ConstructASCIIGeometry(T* dataType1, TString containerName, U*)
                                         ~~~^~~~~~~~~
```

* FairRoot dev branch
* Added `-Wunused-but-set-parameter` in `cmake/modules/CheckCompiler.cmake`
* FairSoft's `test/env/fairroot_develop`
* Debian-10

### PR DESCRIPTION
**Pull request by dennisklein**
Proposal to fix #1145

~~Had to use different function name, because the existing helper's default arguments would have overlapped with my proposed "overload".~~

edit: The original approach grew into hundreds of lines of code changes, so I aborted it (due to lack of test coverage) and went now with the minimal fix I can think of.

### CODE REVIEW

File: base/sim/FairModule.h [hunk 1]
```diff
@@ -168,51 +179,59 @@ class FairModule : public TNamed
...
-    TList* volList = MGeo->getListOfVolumes();
-    // store geo parameter
-    FairRun* fRun = FairRun::Instance();
-    FairRuntimeDb* rtdb = FairRun::Instance()->GetRuntimeDb();
+template<typename GeoModule, typename ParameterSet>
+void FairModule::AddGeoNodesToParameterSetHelper(GeoModule& geoModule, std::string const& parameterSetName)
+{
+    auto const run = FairRun::Instance();
+    auto const rtdb = run->GetRuntimeDb();
+    auto parameterSet = static_cast<ParameterSet*>(rtdb->getContainer(parameterSetName.c_str()));
```
Comment (dennisklein) [inline base/sim/FairModule.h:201]:
```suggestion
    static_assert(std::is_base_of_v<FairParSet, ParameterSet>, "...");
    auto parameterSet = dynamic_cast<ParameterSet*>(rtdb->getContainer(parameterSetName.c_str()));
```

File: base/sim/FairModule.h [hunk 2]
```diff
@@ -168,51 +179,59 @@ class FairModule : public TNamed
...
     }
+    return geoModule;
+}
 
-    TList* volList = MGeo->getListOfVolumes();
-    // store geo parameter
-    FairRun* fRun = FairRun::Instance();
-    FairRuntimeDb* rtdb = FairRun::Instance()->GetRuntimeDb();
+template<typename GeoModule, typename ParameterSet>
+void FairModule::AddGeoNodesToParameterSetHelper(GeoModule& geoModule, std::string const& parameterSetName)
```
Comment (dennisklein) [inline base/sim/FairModule.h:197]:
```suggestion
template<typename ParameterSet, typename GeoModule>
void FairModule::AddGeoNodesToParameterSetHelper(std::string const& parameterSetName, GeoModule& geoModule)
```

Comment (dennisklein) [reply base/sim/FairModule.h:197]: Ideally, it should also be `GeoModule const& geoModule` parameter, but this requires `getListOfVolumes()` to be `const` as well

File: base/sim/FairModule.h [hunk 3]
```diff
@@ -90,9 +90,14 @@ class FairModule : public TNamed
     /** Finish worker run (used in MT mode only) */
     virtual void FinishWorkerRun() const { ; }
 
-    /**template function to construct geometry. to be used in derived classes.*/
+    /**template function to construct geometry. to be used in derived classes.
+     * The first and the third argument are meaningless, just pass nullptr */
```
Comment (ChristianTackeGSI) [inline base/sim/FairModule.h:94]: Can you add a `\deprecated` so it shows up in Doxygen's deprecation list?

**Review by ChristianTackeGSI — APPROVED** [review]

**Review by karabowi — DISMISSED** [review]

File: base/sim/FairModule.h (hunk 3 above)
Comment (dennisklein) [reply base/sim/FairModule.h]: done

**Review by ChristianTackeGSI — APPROVED** [review]

**Review by karabowi — APPROVED** [review]

**Pull request merged**